subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -69,7 +69,10 @@ static void handle_host_cmds_entry(void *arg1, void *arg2, void *arg3)
 		k_sem_give(rx.dev_owns);
 
 		/* Wait until and RX messages is received on host interace */
-		k_sem_take(rx.handler_owns, K_FOREVER);
+		if (k_sem_take(rx.handler_owns, K_FOREVER) < 0) {
```

Comment:
Based on the documentation and waiting K_FOREVER, I didn't find a scenario where this could actually return anything other than 0 (it would instead always just wait forever).

I am still okay with this change though

Reply:
I've updated the PR to add a comment to indicate that the path will never be taken. Let me know if the wording should be changed up!

PS: During a discussion about this issue, we were surprised to find a [bug](https://github.com/zephyrproject-rtos/zephyr/issues/29244) where k_sem_take with K_FOREVER could be coerced to return EAGAIN - the error code reserved for timing out. The bug seems to be in k_thread_resume, though, not k_sem_take, and it happens under a strange condition. So I agree with what you said - there should be no way for your call to return anything but 0.

Reply:
Nice find, and thanks for the added comment!